File: 1993-sum-of-all-subset-xor-totals/1993-sum-of-all-subset-xor-totals.py
class Solution:
    def subsetXORSum(self, nums: List[int]) -> int:
        subset=[]
        sum_ans=0

        for i in range(len(nums)+1):
            for x in itertools.combinations(nums,i):
                subset.append(list(x))

        for sub in subset:
            temp=sub
            sum_temp=0
            for x in temp:
                sum_temp ^=x

            sum_ans+=sum_temp

        return sum_ans

        

File: 1993-sum-of-all-subset-xor-totals/1993-sum-of-all-subset-xor-totals.java
class Solution {
    public int subsetXORSum(int[] nums) {
        return onway(nums, 0, 0);
    }

    public int onway(int[] nums, int xor, int idx) {
        if (idx == nums.length) {
            return xor;
        }
        return onway(nums, xor ^ nums[idx], idx + 1) + onway(nums, xor, idx + 1);
    }
}